### PR TITLE
Fix persistence of webconsole prefs

### LINKED ISSUE #5845: Console timestamps are shown by default
**Description:**
I bet @markerikson recent console pref change removed a pref which was storing console timestamps

Bug: console logs now show timestamps by default
ER: the pref is off by default and persisted

### PR DESCRIPTION
**Pull request by markerikson**
This PR:

- Adds `consoleUI` to the `UIState` definition
- Restores use of `prefs.timestampsVisible` in defining `consoleUI` initial state
- Restructures webconsole prefs persistence to centralize the persistence logic and actually set persistence values reactively

Previously, the "log level" prefs were being persisted in a thunk,
in `actions/filters`.  Jason wants pref persistence to be reactive
and in a single location, in `ui/setup/prefs`.

Also, the `timestampsVisible` flag wasn't being persisted properly,
and the `node_modules` pref looks like it was never persisted.  Both
of those are now persisted correctly.

Fixes #5845

### CODE REVIEW

File: src/devtools/client/webconsole/reducers/ui.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 import { createSlice, createNextState } from "@reduxjs/toolkit";
 
-interface WebconsoleUIState {
+export interface WebconsoleUIState {
```
Comment (jaril) [inline src/devtools/client/webconsole/reducers/ui.ts:6]: This is the smallest nit but what's your take on camel case for acronyms — WebconsoleUIState, or WebconsoleUiState?

File: src/ui/setup/prefs.ts [hunk 2]
```diff
@@ -37,20 +38,48 @@ export function registerStoreObserver(
...
   }
 
   updatePref("viewMode", getViewMode);
+
   updateAsyncPref("eventListenerBreakpoints", (state: UIState) => state.eventListenerBreakpoints);
   updateAsyncPref("commandHistory", (state: UIState) => state.messages?.commandHistory);
 
+  // This is lazy-loaded, so it may not exist on startup
+  if (state.consoleUI && oldState.consoleUI) {
+    console.log(state);
```
Comment (jaril) [inline src/ui/setup/prefs.ts:68]: :salt:

File: src/ui/setup/prefs.ts [hunk 3]
```diff
@@ -37,20 +38,48 @@ export function registerStoreObserver(
...
+      return state.consoleUI.timestampsVisible;
+    });
+  }
+
+  if (state.filters && oldState.filters) {
+    updateWebconsolePref("filterError", (state: UIState) => state.filters.error);
+    updateWebconsolePref("filterWarn", (state: UIState) => state.filters.warn);
+    updateWebconsolePref("filterInfo", (state: UIState) => state.filters.info);
+    updateWebconsolePref("filterDebug", (state: UIState) => state.filters.debug);
+    updateWebconsolePref("filterLog", (state: UIState) => state.filters?.log);
```
Comment (jaril) [inline src/ui/setup/prefs.ts:79]: Is there something special about log that we need the optional chaining in `state.filters?.log`?

**Review by jaril — APPROVED** [review]
Left non-blocking comments. Just need to take out the `console.log` and we're good to go!

File: src/devtools/client/webconsole/reducers/ui.ts (hunk 1 above)
Comment (markerikson) [reply src/devtools/client/webconsole/reducers/ui.ts:6]: Uh... I've never had to formally think about this before? :)

https://stackoverflow.com/questions/15526107/acronyms-in-camelcase looks like it has some good discussion

I've fallen towards `userId` as of late, although "ID" isn't an acronym.

Microsoft's advice suggests capitalizing both letters in a two-letter acronym.

But no hugely strong opinion here.

**Pull request merged**